core/src/sql/value/into_json.rs

```diff
@@ -0,0 +1,549 @@
+			// the client.
+			Value::None | Value::Null => JsonValue::Null,
+			Value::Bool(boolean) => boolean.into(),
+			Value::Number(number) => match number {
+				Number::Int(int) => int.into(),
+				Number::Float(float) => float.into(),
+				Number::Decimal(decimal) => json!(decimal),
+			},
+			Value::Strand(strand) => strand.0.into(),
+			Value::Duration(duration) => duration.to_raw().into(),
+			Value::Datetime(datetime) => json!(datetime.0),
+			Value::Uuid(uuid) => json!(uuid.0),
+			Value::Array(array) => JsonValue::Array(Array::from(array).0),
+			Value::Object(object) => JsonValue::Object(Object::from(object).0),
+			Value::Geometry(geo) => Geometry::from(geo).0,
+			Value::Bytes(bytes) => json!(bytes.0),
+			Value::Thing(thing) => thing.to_string().into(),
+			// These Value types are un-computed values
+			// and are not used in query responses sent
+			// to the client.
+			Value::Param(param) => json!(param),
+			Value::Idiom(idiom) => json!(idiom),
+			Value::Table(table) => json!(table),
+			Value::Mock(mock) => json!(mock),
+			Value::Regex(regex) => json!(regex),
+			Value::Block(block) => json!(block),
+			Value::Range(range) => json!(range),
+			Value::Edges(edges) => json!(edges),
+			Value::Future(future) => json!(future),
+			Value::Constant(constant) => match constant.value() {
+				ConstantValue::Datetime(datetime) => json!(datetime.0),
+				ConstantValue::Float(float) => float.into(),
+			},
+			Value::Cast(cast) => json!(cast),
+			Value::Function(function) => json!(function),
+			Value::Model(model) => json!(model),
+			Value::Query(query) => json!(query),
+			Value::Subquery(subquery) => json!(subquery),
+			Value::Expression(expression) => json!(expression),
+		}
+	}
+}
+
+#[derive(Serialize)]
+struct Array(Vec<JsonValue>);
+
+impl From<sql::Array> for Array {
+	fn from(arr: sql::Array) -> Self {
+		let mut vec = Vec::with_capacity(arr.len());
+		for value in arr {
+			vec.push(value.into());
+		}
+		Self(vec)
+	}
+}
+
+#[derive(Serialize)]
+struct Object(Map<String, JsonValue>);
+
+impl From<sql::Object> for Object {
+	fn from(obj: sql::Object) -> Self {
+		let mut map = Map::with_capacity(obj.len());
+		for (key, value) in obj {
+			map.insert(key.to_owned(), value.into());
+		}
+		Self(map)
+	}
+}
+
+#[derive(Serialize)]
+enum CoordinatesType {
+	Point,
+	LineString,
+	Polygon,
+	MultiPoint,
+	MultiLineString,
+	MultiPolygon,
+}
+
+#[derive(Serialize)]
+struct Coordinates {
+	#[serde(rename = "type")]
+	typ: CoordinatesType,
+	coordinates: JsonValue,
+}
+
+struct GeometryCollection;
+
+impl Serialize for GeometryCollection {
+	fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		s.serialize_str("GeometryCollection")
+	}
+}
+
+#[derive(Serialize)]
+struct Geometries {
+	#[serde(rename = "type")]
+	typ: GeometryCollection,
+	geometries: Vec<JsonValue>,
+}
+
+#[derive(Serialize)]
+struct Geometry(JsonValue);
+
+impl From<sql::Geometry> for Geometry {
```

Comment:
It seems like the `geo` crate has an optional `serde` feature. Should we depend on that instead of our own custom logic?

Reply:
It does and we already do. The code in this module is not for serialisation but for converting `surrealdb::sql::Value` to `serde_json::Value`. Most of it is already on `main`. It's just moved here from `sql::value::serde::de` because we no longer user JSON as an intermediary format for deserialisation.